java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -268,7 +274,22 @@ ArrowBuf readBuffer(BufferAllocator allocator, int count) throws IOException {
     }
   }
 
+  private abstract class DataBufferReader {
```

Comment:
Why do we need separate abstract/impl classes? Just merge them

Reply:
Would it be okay for BufferReader interface to work with `List<ArrowBuf>` instead or `ArrowBuf`

Reply:
Otherwise, we still have to do a type check below because if I keep `readBuffer` for non list and `readBuffers` for list, I need to do type check for calling the right function. That's why.

Reply:
? 

Don't make inheritance hierarchies that don't need to exist. This whole thing could just be a method or two.